docs/firebase_in_libraries.md

```diff
@@ -1,31 +1,28 @@
-# Using Firebase from a Framework or a library
+# Using Firebase from a framework or a library
 
-Currently all source code and binary Firebase SDKs are compiled as **static**
-frameworks. It may be not too important detail if Firebase SDKs are used from an
-app target directly, but must be taken into account if Firebase is used from
-another library or a framework. The key difference is in the way static and
-dynamic linking works. Your framework itself may be either static or dynamic.
-Let's consider these two options in more details.
+Currently all source code and binary Firebase SDKs are compiled as static frameworks. Most
+of the time you’ll link these frameworks directly to your app targets, but in some cases
+it makes sense to link Firebase frameworks to your app indirectly, from another library or
+another framework. This note talks about some of the pitfalls of this technique, almost
+always hard-to-debug undefined behaviors, that all come down to code duplication and how
+static and dynamic linking work. Your framework itself may be either static or dynamic. Let's
+consider these two options in more detail.
 
-## Using Firebase SDKs from dynamic framework
+## Using Firebase SDKs from dynamic frameworks
 
-When a dynamic framework/library is compiled, each static dependency symbols are
-added to the dynamic framework binary. It is not really important unless you
-need to use both the dynamic framework and one of its static dependencies
-directly in your app or another framework. In this case if you attempt to link
-the same static framework/library to both the app and the dynamic framework you
-will end up with the static framework symbols added to both the app and the
-dynamic framework binaries. It means that when you app launches it will have two
-copy of the static framework symbols. This leads to undefined behavior
-(especially when different versions of the static framework are linking to the
-app and the dynamic framework). For example, a `dispatch_once` may or may not do
-the right initialization since there are now two entities to initialize. Here
-are couple more examples of issues related to this undefined behavior:
+A dynamic framework is a bundle containing dynamic libraries and other resources. The dynamic
+libraries bundled in the framework can themselves be compiled from static libraries, like
+Firebase core and product libraries, meaning all of the symbols of the static libraries are
+part of the dynamic framework bundle. What if your core app already directly links a static
+Firebase library when you link to the same library indirectly from a dynamic framework? This
+leads to undefined behavior (especially when different versions of the static framework are
```

Comment:
From the first glance it doesn't look exactly clear that the undefined behaviour is a result two(or more) copies of symbols are present in its runtime when such an application is launched. Do you think we should/can reword it to highlight this fact?

Reply:
Thanks!